src/MOIWrapper.jl

```diff
@@ -0,0 +1,397 @@
+end
+
+
+function MOI.optimize!(cbcOptimizer::CbcOptimizer)
+    # Call solve function
+    CbcCI.solve(cbcOptimizer.inner)
+end
+
+
+
+## canadd, canset, canget functions
+
+function MOI.canaddvariable(cbcOptimizer::CbcOptimizer)
+    return false
+end
+
+function MOI.cantransformconstraint(model::CbcOptimizer, c::MOI.ConstraintIndex{F}, ::Type{S}) where {F<:MOI.AbstractFunction, S<:MOI.AbstractSet}
+    return false
+end
+
+## supports constraints
+
+function MOI.supportsconstraint(cbcOptimizer::CbcOptimizer, F::Type{<:MOI.AbstractFunction}, S::Type{<:MOI.AbstractScalarSet})
+    return (F,S) in SUPPORTED_CONSTRAINTS
+end
+
+function MOI.supports(cbcOptimizer::CbcOptimizer, object::MOI.ObjectiveFunction{T}) where T <: MOI.AbstractFunction
+    return T in SUPPORTED_OBJECTIVES
+end
+
+
+
+## Set functions
+
+
+# empty!
+
+function MOI.empty!(cbcOptimizer::CbcOptimizer)
+    cbcOptimizer = CbcOptimizer()
+end
+
+
+function MOI.set!(cbcOptimizer::CbcOptimizer, object::MOI.ObjectiveSense, sense::MOI.OptimizationSense)
+    sense == MOI.MinSense && CbcCI.setObjSense(cbcOptimizer.inner, 1)
+    sense == MOI.MaxSense && CbcCI.setObjSense(cbcOptimizer.inner, -1)
+end
+
+
+## Get functions
+
+const CANGETOBJECTS = [
+    MOI.TerminationStatus,
+    MOI.NodeCount,
+    MOI.ObjectiveValue,
+    MOI.ObjectiveBound,
+    MOI.ObjectiveSense,
+    MOI.ResultCount
+]
+
+const CANGETVARATTR = [
+    MOI.VariablePrimal
+]
+const CANGETCONSATTR = []
+
+function MOI.canget(cbcOptimizer::CbcOptimizer, object::MOI.PrimalStatus)
+    object.N >= 2 && return false
+    return MOI.get(cbcOptimizer, MOI.ResultCount()) == 1
+end
+
+
+function MOI.canget(cbcOptimizer::CbcOptimizer, object::Union{MOI.AbstractModelAttribute, MOI.AbstractOptimizerAttribute})
+    return typeof(object) in CANGETOBJECTS
+end
+
+function MOI.canget(cbcOptimizer::CbcOptimizer, object::MOI.AbstractConstraintAttribute, indexType::Type{MOI.ConstraintIndex})
+    return typeof(object) in CANGETCONSATTR
+end
+
+function MOI.canget(cbcOptimizer::CbcOptimizer, object::MOI.AbstractVariableAttribute, indexType::Type{MOI.VariableIndex})
+    return typeof(object) in CANGETVARATTR
+end
+
+function MOI.canget(cbcOptimizer::CbcOptimizer, object::MOI.AbstractVariableAttribute, index::MOI.VariableIndex)
+    return typeof(object) in CANGETVARATTR
+end
+
+
+# isempty
+function MOI.isempty(cbcOptimizer::CbcOptimizer)
+    return (CbcCI.getNumCols(cbcOptimizer.inner) == 0 && CbcCI.getNumRows(cbcOptimizer.inner) == 0)
+end
+
+function MOI.get(cbcOptimizer::CbcOptimizer, object::MOI.VariablePrimal, ref::MOI.VariableIndex)
+    variablePrimals = CbcCI.getColSolution(cbcOptimizer.inner)
+    return variablePrimals[ref.value]
+end
+
+function MOI.get(cbcOptimizer::CbcOptimizer, object::MOI.VariablePrimal, ref::Vector{MOI.VariableIndex})
+    variablePrimals = CbcCI.getColSolution(cbcOptimizer.inner)
+    askedPrimals = Vector{Float64}()
```

Comment:
Just use generator syntax instead of creating an empty array then looping:
`return [variablePrimals[vi.value] for vi in ref]` or
`return map(vi->variablePrimals[vi.value], ref)`